### PR TITLE
1.4.0: MobilePerceivedObject RSU computation + SPATEM/MAPEM

### PR DESCRIPTION
**Pull request by nbuffon**
What's new ?
------------------

* Message no longer implements Mobile trait
* `as_mobile()` method added to Message to access Mobile when manipulating generic Message
* SPATEM struct
* MAPEM struct
* Extend PerceivedObject management to RSU originating CPM
* Bump version to 1.4.0

How to test
---------------
1. Build and test the lib and examples
    ```
    cargo clean; cargo build --all; cargo test --all
    ```
    **=> Build must success**  
    **=> All tests must pass**  

2. Start local mosquitto server
    ```
    systemctl start mosquitto
    ```

3. Start its-client
    ```
    cargo run --bin its-client -- --mqtt-client-id unique_id_111 --root-topic 5GCroCo/inQueue
    ```

4. Start the Paul Sabatier's single vehicle SUMO scenario
    ```
    ./start.sh -H localhost -P 1883 --mqtt-client-id sumo_ps_nbu -c ../sumo-paul-sabatier/sumo_data/static_vehicle.sumocfg
    ```
    **=> On the HMI you must see two vehicles instead of one (the second being a 3 seconds delayed copy of the first one)**

### CODE REVIEW

**Comment by tigroo** [comment]
rustfmt to pass before to merge.

**Comment by tigroo** [comment]
`cargo build --all` instead of `carbo build --all` :D

**Comment by tigroo** [comment]
test failed: no more copycat cam created. I'm searching why...

File: rust/libits-client/src/reception/exchange/mobile_perceived_object.rs [hunk 1]
```diff
@@ -100,7 +112,7 @@ fn compute_id(object_id: u8, cpm_station_id: u32) -> u32 {
     }
 }
 
-fn compute_position(
+fn compute_position_from_mobile(
```
Comment (tigroo) [inline rust/libits-client/src/reception/exchange/mobile_perceived_object.rs:115]: why to rename?

File: rust/libits-client/src/reception/exchange/mobile_perceived_object.rs [hunk 2]
```diff
@@ -118,26 +130,78 @@ fn compute_speed(x_speed: i16, y_speed: i16) -> u16 {
     speed_from_yaw_angle(x_speed, y_speed)
 }
 
-fn compute_heading(y_distance: i32, cpm_heading: u16) -> u16 {
-    match y_distance {
-        y if y < 0 => (cpm_heading + 1800) % 3600,
+fn compute_heading_from_mobile(perceived_object: &PerceivedObject, cpm_heading: u16) -> u16 {
```
Comment (tigroo) [inline rust/libits-client/src/reception/exchange/mobile_perceived_object.rs:133]: compute_heading_from_vehicle instead?

**Review by tigroo — CHANGES_REQUESTED** [review]

**Comment by tigroo** [comment]
no way to use the HMI on localhost,no? I switch on the internet instances

**Comment by tigroo** [comment]
un the internet instance, it seems the copycat cam doesn't use different source uuid : all have got the same name on the HMI :( (I used sumo-paul-sabatier/sumo_data/two_vehicles_jam.sumocfg)

File: rust/libits-client/src/reception/exchange/message.rs [hunk 3]
```diff
@@ -21,8 +23,10 @@ use crate::reception::typed::Typed;
 #[serde(untagged)]
 pub enum Message {
     CAM(CooperativeAwarenessMessage),
-    DENM(DecentralizedEnvironmentalNotificationMessage),
     CPM(CollectivePerceptionMessage),
+    DENM(DecentralizedEnvironmentalNotificationMessage),
+    MAPEM(MAPExtendedMessage),
```
[the remark this answers is not in the document]
Comment (nbuffon) [reply rust/libits-client/src/reception/exchange/message.rs:28]: MAP is no acronym here, it stands for map literally 

From ETSI's ITS spec: 
> MAPEM MAP (topology) Extended Message

File: THIRD-PARTY.md [hunk 4]
```diff
@@ -18,6 +18,9 @@
 #### flexi_logger
 - [Source code](https://github.com/emabee/flexi_logger)
 
+#### geo
```
[the remark this answers is not in the document]
Comment (nbuffon) [reply THIRD-PARTY.md:21]: Ah I missed it, I was glad that I thought about adding it here then :/

**Comment by nbuffon** [comment]
> no way to use the HMI on localhost,no? I switch on the internet instances

Actually you can, you just have to listen to `inQueue` instead of `outQueue`

File: rust/libits-client/src/reception/exchange/mobile_perceived_object.rs (hunk 2 above)
Comment (nbuffon) [reply rust/libits-client/src/reception/exchange/mobile_perceived_object.rs:133]: I can change `mobile` for `vehicle` in the signatures    
I can be relevant as it is originating from some "vehicle_container"

File: rust/libits-client/src/reception/exchange/mobile_perceived_object.rs (hunk 1 above)
Comment (nbuffon) [reply rust/libits-client/src/reception/exchange/mobile_perceived_object.rs:115]: For consistency, as all vehicle relative computation methods are suffixed the same to make it explicit about what they do

**Pull request merged**